biosys/apps/main/api/uploaders.py

```diff
@@ -194,7 +194,8 @@ def _create_record(self, row, counter):
                 tz = self.dataset.project.timezone or timezone.get_current_timezone()
                 record.datetime = timezone.make_aware(observation_date, tz)
                 # geometry
-                geometry = self.schema.cast_geometry(row, default_srid=MODEL_SRID)
+                default_srid = self.dataset.project.datum or MODEL_SRID
+                geometry = self.schema.cast_geometry(row, default_srid=default_srid)
```

Comment:
Maybe here do like line 141 of validators.py

i.e.: `self.schema.cast_geometry(data, default_srid=self.default_srid or MODEL_SRID)`